optimum/intel/ipex/modeling_base.py

```diff
@@ -259,16 +284,17 @@ def forward(
         if "attention_mask" in self.input_names:
             inputs["attention_mask"] = attention_mask
 
-        outputs = self.model(**inputs)
+        outputs = self._call_model(**inputs)
         return ModelOutput(**outputs) if isinstance(outputs, dict) else ModelOutput(logits=outputs[0])
 
 
 class IPEXModelForQuestionAnswering(IPEXModel):
     auto_model_class = AutoModelForQuestionAnswering
     export_feature = "question-answering"
 
+    @wraps(IPEXModel.forward)
     def forward(self, *args, **kwargs):
-        outputs = self.model(*args, **kwargs)
+        outputs = super().forward(*args, **kwargs)
```

Comment:
why is this needed ?

Reply:
The `prepare_jit_inputs` looks at the signature of the function and the wraps and super help avoid code copy

Reply:
would prefer we avoid as it will fail in case `outputs` is not a dict https://github.com/huggingface/optimum-intel/blob/8ee487dc2ade5bd0023d1bbe0a0103d6af8821e0/optimum/intel/ipex/modeling_base.py#L193

Reply:
also not sure to see the link with `prepare_jit_inputs`

Reply:
1. In `_init_warmup` we call `prepare_jit_inputs` which examines the passed model's `forward` signature to see which dummy inputs exists in the signature. If we don't use wraps we get the signature of 
    ```
    self, *args, **kwargs
    ```
     instead of 
    ```
    self, input_ids: torch.Tensor, attention_mask: torch.Tensor, token_type_ids: torch.Tensor = None, **kwargs,
    ```
2. `outputs` will always be a dict because this is the output of `IPEXModel.forward`, no?

Reply:
1. OK I understand, was thinking that `prepare_jit_inputs` was only used for the torchscript export but I see that it's also used in `_init_warmup`, thanks for the clarification
2. here I'm talking about outputs https://github.com/huggingface/optimum-intel/blob/8ee487dc2ade5bd0023d1bbe0a0103d6af8821e0/optimum/intel/ipex/modeling_base.py#L192C9-L192C16